src/components/DetailsChart/index.tsx

```diff
@@ -11,11 +11,34 @@ interface DetailsChartProps {
   poolHistory: { data: LiquidityPoolHistory[] };
 }
 
+const findSegmentBoundaryIndex = (
+  data: LiquidityPoolHistory[],
+  daysAgo: number,
+) => {
+  /* iterate over poolHistory.data and find the first timestamp 
+  that corresponds to the relative date we're looking. 
+  For ex, for 30 days ago, find which timestamp is ~ 30 days ago */
+
+  const today = new Date();
+
+  // We want to find the first date that is newer than daysAgo + 1
+  const priorDateTs = new Date().setDate(today.getDate() - daysAgo + 1);
+
+  const index = data.findIndex((datum) => datum.ts < priorDateTs);
```

Comment:
❓ not too familiar on the pool history dataset, but to be sure is the pool history data sorted starting from current date to older dates? (if I'm understanding this line correctly)

Reply:
Yup, that's exactly right. The most first data entry has the most recent timestamp